### PR TITLE
Explicit header whitelist in configuration

### LINKED ISSUE #389: 🚀 Configurable :request_headers / APPSIGNAL_REQUEST_HEADERS
Depends on decisions made in https://github.com/appsignal/appsignal-elixir/issues/317. As requested via [support](https://app.intercom.io/a/apps/yzor8gyw/inbox/inbox/unassigned/conversations/14994864230) (private Intercom link).

### PR DESCRIPTION
**Pull request by jeffkreeftmeijer**
Adds an explicit header whitelist to the configuration file or environment variables created/suggested by the installer. Closes #389.

This prevents AppSignal from unintentionally storing user identifiable information by default from headers. Use the new whitelist configuration option to specify for your apps which headers AppSignal should store. The default list is a minimal list which you can expand in the appsignal.yml config file or in the system environment variables.

For more information see our docs: https://docs.appsignal.com/ruby/configuration/ 

## TODO

- [x] Add a warning if no request headers are configured (https://github.com/appsignal/appsignal-elixir/pull/327#discussion_r180010430 )

*Note*: This is a work in progress until we've decided on the final list of default headers.

### CODE REVIEW

**Comment by tombruijn** [comment]
@jeffkreeftmeijer updated the header list and added the warning to the log when the option is not configured. Now ready for review?

**Comment by jeffkreeftmeijer** [comment]
Yes! I'm 👍 on your changes. :)

File: lib/appsignal/config.rb [hunk 1]
```diff
@@ -189,10 +197,35 @@ def validate
...
+    def check_if_request_headers_option_is_set
+      return unless valid?
+      return if config_hash.key?(:request_headers) || env == "test"
+
+      multi_line_request_headers = SUGGESTED_REQUEST_HEADERS.map do |row|
+        row.map(&:inspect).join(", ")
+      end.join(",\n    ")
+      single_line_request_headers = SUGGESTED_REQUEST_HEADERS.flatten.join(",")
+
+      logger.warn "The `request_headers` config option was not set " \
```
Comment (tombruijn) [inline lib/appsignal/config.rb]: @jeffkreeftmeijer wants to log this to STDOUT, so we're going to change it. This will make it stand out more.

**Comment by jeffkreeftmeijer** [comment]
Can't review, but I'm :+1: on the STDOUT changes. :shipit:

**Comment by tombruijn** [comment]
Let's merge once build is green.

**Comment by tombruijn** [comment]
Additional happy side effect I just noticed: This change also removes any remaining warnings that were printed in the test suite output. 😁 pinging #299

**Pull request merged**